modules.cmake

```diff
@@ -203,6 +213,44 @@ function(add_module_library name)
     endforeach ()
   endif ()
 
+
+  if (CMAKE_COMPILER_IS_GNUCXX)
+    get_target_property(std ${name} CXX_STANDARD)
+
+    # Add .gcm/.o files as sources to make sure they are built before the library.
+    set(sources_in ${sources})
+    set(sources)
+    set(gcms)
+    foreach (src ${sources_in})
+      get_filename_component(src_we ${src} NAME_WE)
+      set(obj ${src_we}.o)
+      set(sources ${sources} ${CMAKE_CURRENT_BINARY_DIR}/${obj})
+      set(gcm ${CMAKE_CURRENT_BINARY_DIR}/gcm.cache/${src_we}.gcm)
+      set(gcms ${gcms} ${gcm})
+      add_custom_command(
+        OUTPUT ${obj} ${gcm}
+        COMMAND ${CMAKE_CXX_COMPILER} $<TARGET_PROPERTY:${name},COMPILE_OPTIONS>
+        -std=c++${std} -fmodules-ts
+        -xc++ -c -o ${obj} ${src}
+        # Required by the generator expression above.
+        COMMAND_EXPAND_LISTS
+        DEPENDS ${src})
+    endforeach ()
+
+    # Generate moduler mapper `-fmodule-mapper`
+    add_custom_command(
+      OUTPUT ${mapper_file}
+      COMMAND ${CMAKE_COMMAND} -DCMAKE_CURRENT_BINARY_DIR=${CMAKE_CURRENT_BINARY_DIR} -DMAPPER_FILE=${mapper_file} -P ${CMAKE_CURRENT_LIST_DIR}/../mapper.cmake
+      # Required by the generator expression above.
+      COMMAND_EXPAND_LISTS
+      DEPENDS ${gcms})
+
+    # add dummy target to handle updates
+    add_custom_target(${name}_dummy__ DEPENDS
+      ${gcms} ${mapper_file})
+    add_dependencies(${name} ${name}_dummy__)
```

Comment:
Is an extra target needed? Can the dependency be added to `${name}` directly?

Reply:
Done